### ISSUE TITLE
 git-latexdiff like functionality

### ISSUE THREAD

**Issue by fangrh** [typst/webapp-issues#55]

**Comment by baptiste** [comment]
I agree it would be _super useful_ to have a tool like latexdiff (and hopefully better at handling things like bibtex, figures, etc.).

My probably naive way of thinking about this problem would be:

1. We have two ASTs. Comparing trees and merging them probably has a standard solution in Rust that can be reused. 

2. The changes need to be identified with a start and end point, so the content in-between can be styled appropriately (red/blue, or more fancy background, strikethrough, etc.). The [pinit package](https://github.com/typst/packages/tree/main/packages/preview/pinit/0.1.1) seems to have a neat interface for this (I don't know if it can handle all cases robustly enough, but it looks promising).
